akka-contrib/src/main/scala/akka/contrib/pattern/DistributedPubSubMediator.scala

```diff
@@ -548,6 +574,10 @@ class DistributedPubSubMediator(
       content = bucket.content + (key -> ValueHolder(v, valueOption))))
   }
 
+  def getCurrentTopics(): Map[String, ActorRef] = {
+    context.children.map(ref ⇒ (URLDecoder.decode(ref.path.name, "utf-8"), ref)).toMap
+  }
```

Comment:
The registry contains one bucket per node, we iterate all of those.

Each bucket contains ActorRef of the actors that have been registered with `Put` and also child ActorRef for topics that have been registered with `Subscribe`. We are only interested in the latter here. Those are are keyed by their path without address, i.e. the subscribe topics all start with the same string as the `self` (DistributedPubSubMediator actor) path. That is why we filter out only topics by looking at the prefix.

Then there is a special kind of topics for the `group` feature. We exclude those also (`!topic.contains('/') // exclude group topics`).

Remaining should be all topics that have been registered with `Subscribe`.